Plugins/NativeCamera/Source/Android/NativeCameraImpl.cpp

```diff
@@ -51,35 +65,92 @@ namespace Babylon::Plugins
         return oesTexture;
     }
 
-    std::string Camera::Impl::GetCameraId(bool frontCamera)
+    Camera::Impl::CameraConfiguration Camera::Impl::GetCameraConfiguration(uint32_t maxWidth, uint32_t maxHeight, bool frontCamera)
     {
+        // Get the list of available cameras
         API24::ACameraIdList *cameraIds = nullptr;
         GET_CAMERA_FUNCTION(ACameraManager_getCameraIdList)(m_cameraManager, &cameraIds);
 
-        std::string cameraId{};
+        const char* bestCameraId{nullptr};
+        uint32_t bestPixelCount{0};
+        uint32_t bestDimDiff{0};
+        uint32_t bestWidth{0};
+        uint32_t bestHeight{0};
+        int32_t bestSensorOrientation{0};
+        bool foundExactMatch{false};
 
+        // Iterate over all of the cameras and find the one that has the best stream configuration
         for (int i = 0; i < cameraIds->numCameras; ++i)
         {
-            const char *id = cameraIds->cameraIds[i];
+            const char* id = cameraIds->cameraIds[i];
 
             API24::ACameraMetadata *metadataObj;
             GET_CAMERA_FUNCTION(ACameraManager_getCameraCharacteristics)(m_cameraManager, id, &metadataObj);
 
             API24::ACameraMetadata_const_entry lensInfo = {};
             GET_CAMERA_FUNCTION(ACameraMetadata_getConstEntry)(metadataObj, API24::ACAMERA_LENS_FACING, &lensInfo);
 
+            API24::ACameraMetadata_const_entry sensorOrientation = {};
+            GET_CAMERA_FUNCTION(ACameraMetadata_getConstEntry)(metadataObj, API24::ACAMERA_SENSOR_ORIENTATION, &sensorOrientation);
+
             auto facing = static_cast<API24::acamera_metadata_enum_android_lens_facing_t>(lensInfo.data.u8[0]);
+            if (facing != (frontCamera ? API24::ACAMERA_LENS_FACING_FRONT : API24::ACAMERA_LENS_FACING_BACK))
+            {
+                // Ignore cameras facing the wrong direction
+                continue;
+            }
+
+            // Get all available stream configurations supported by the camera
+            API24::ACameraMetadata_const_entry streamConfigurations;
```

Comment:
Should this match the others initialization?
```suggestion
            API24::ACameraMetadata_const_entry streamConfigurations = {};
```